src/main/java/mil/dds/anet/database/PositionDao.java

```diff
@@ -170,134 +167,204 @@ public int updateInternal(Position p) {
+        .execute();
+  }
+
+  @InTransaction
+  public int removePersonFromPosition(String positionUuid) {
+    int numRows = 0;
+    Instant now = Instant.now();
+    numRows += cleanPosition(positionUuid);
+
+    PersonPositionHistory currentPositionPerson = getDbHandle()
+        .createQuery("/* personEmptyPosition.find */ SELECT * FROM \"peoplePositions\""
+            + " WHERE \"positionUuid\" = :positionUuid AND \"personUuid\" IS NOT NULL AND "
+            + " \"endedAt\" IS NULL")
+        .bind("positionUuid", positionUuid).map(new PersonPositionHistoryMapper()).findFirst()
+        .orElse(null);
+
+    numRows += getDbHandle()
+        .createUpdate(
+            "/* positionRemovePerson.end */ UPDATE \"peoplePositions\" SET \"endedAt\" = :endedAt "
+                + " WHERE \"positionUuid\" = :positionUuid AND \"personUuid\" IS NOT NULL AND "
+                + " \"endedAt\" IS NULL")
+        .bind("positionUuid", positionUuid).bind("endedAt", DaoUtils.asLocalDateTime(now))
+        .execute();
+
+    if (currentPositionPerson != null) {
+      removePositionFromPerson(currentPositionPerson.getPersonUuid());
+    }
+
+    PersonPositionHistory emptyPersonPosition = getDbHandle()
+        .createQuery("/* personEmptyPosition.find */ SELECT * FROM \"peoplePositions\""
+            + " WHERE \"positionUuid\" = :positionUuid "
+            + " AND \"personUuid\" IS NULL AND \"endedAt\" IS NULL")
+        .bind("positionUuid", positionUuid).map(new PersonPositionHistoryMapper()).findFirst()
+        .orElse(null);
+
+    if (emptyPersonPosition == null) {
+      numRows += insertPersonEmptyInPosition(positionUuid);
+    }
+
+    numRows += deletePersonEmptyInPositionEmpty();
+    if (numRows > 0) {
+      return 1;
+    }
+    return numRows;
+  }
+
+  @InTransaction
+  public int removePositionFromPerson(String personUuid) {
+    int numRows = 0;
+    Instant now = Instant.now();
+
+    PersonPositionHistory currentPositionPerson = getDbHandle()
+        .createQuery("/* personEmptyPosition.find */ SELECT * FROM \"peoplePositions\""
+            + " WHERE \"personUuid\" = :personUuid AND \"positionUuid\" IS NOT NULL AND "
+            + " \"endedAt\" IS NULL")
+        .bind("personUuid", personUuid).map(new PersonPositionHistoryMapper()).findFirst()
+        .orElse(null);
+
+    numRows += getDbHandle()
+        .createUpdate(
+            "/* positionRemovePerson.end */ UPDATE \"peoplePositions\" SET \"endedAt\" = :endedAt "
+                + " WHERE \"personUuid\" = :personUuid AND \"positionUuid\" IS NOT NULL AND "
+                + " \"endedAt\" IS NULL")
+        .bind("personUuid", personUuid).bind("endedAt", DaoUtils.asLocalDateTime(now)).execute();
+
+    if (currentPositionPerson != null) {
+      removePersonFromPosition(currentPositionPerson.getPositionUuid());
+    }
+
+    PersonPositionHistory emptyPositionPerson = getDbHandle()
+        .createQuery("/* personEmptyPosition.find */ SELECT * FROM \"peoplePositions\""
```

Comment:
Please ensure that each database statement has a unique comment.

Reply:
Fixed with using unique comments.